src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+
+        var windows = Application.Current?.Windows;
+        if(windows != null)
+        {
+            ApplyTheme(windows , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+        }
+    }
+
+    internal static void ApplySystemTheme(Window window, bool forceUpdate = false)
+    {
+        string systemTheme = GetSystemTheme();
+        bool useLightMode = GetUseLightTheme();
+        Color systemAccentColor = DwmColorization.GetSystemAccentColor();
+        ApplyTheme(new ArrayList { window } , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+    }
+
+    internal static void ApplyTheme(
+        IEnumerable windows, 
+        string requestedTheme, 
+        bool requestedUseLightMode,
+        Color requestedAccentColor, 
+        bool forceUpdate = false)
+    {
+        bool needsUpdate = forceUpdate;
+        
+        if(requestedUseLightMode != _currentUseLightMode || DwmColorization.GetSystemAccentColor() != DwmColorization.CurrentApplicationAccentColor)
+        {
+            DwmColorization.UpdateAccentColors();
+            needsUpdate = true;
+        }
+
+        if(needsUpdate || requestedTheme != _currentApplicationTheme || requestedUseLightMode != _currentUseLightMode)
+        {
+            Uri dictionaryUri = GetFluentWindowThemeResourceUri(requestedTheme, requestedUseLightMode);
+
+            UpdateFluentWindowsThemeResources(dictionaryUri);
+
+            foreach(Window window in windows)
+            {
+                SetImmersiveDarkMode(window, !requestedUseLightMode);
+                window.CoerceValue(Window.WindowBackdropTypeProperty);
+            }
+
+            _currentApplicationTheme = requestedTheme;
+            _currentUseLightMode = requestedUseLightMode;
+        }
+    }
+
+    private static bool SetImmersiveDarkMode(Window window, bool useDarkMode)
+    {
+        if (window == null)
+        {
+            return false;
+        }
+
+        IntPtr handle = new WindowInteropHelper(window).Handle;
+
+        if (handle != IntPtr.Zero)
+        {
+            var dwmResult = NativeMethods.DwmSetWindowAttributeUseImmersiveDarkMode(handle, useDarkMode);
+            return dwmResult == HRESULT.S_OK;
+        }
+
+        return false;
+    }
+
+    #region Helper Methods
+
+    internal static string GetSystemTheme()
+    {
+        string systemTheme = Registry.GetValue(_regThemeKeyPath,
+            "CurrentTheme", "aero.theme") as string ?? String.Empty;
+
+        return systemTheme;
+    }
+   
+    internal static bool GetUseLightTheme()
+    {
+        var appsUseLightTheme = Registry.GetValue(_regPersonalizeKeyPath,
+            "AppsUseLightTheme", null) as int?;
+
+        if (appsUseLightTheme == null)
+        {
+            return Registry.GetValue(_regPersonalizeKeyPath,
+                "SystemUsesLightTheme", null) as int? == 0 ? false : true;
```

Comment:
Pretty sure that if it returns `null`, it will not equal to zero and you will return `true`. Same issue on the line with `return`.

```C#
var useLightTheme = ..."AppUseLightTheme"... as int?;
if (useLightTheme == null)
    useLightTheme = ..."SystemUsesLightTheme"... as int?;
return useLightTheme == 1; // or useLightTheme.GetValueOrDefault() != 0;
```

Reply:
Actually, if we are returning null, then that means we don't have the key right.
So, in that case, we have to choose one default. That is what I am doing in lin 131-132.

For the other one, we already check if the value is null. So I think, we are okay there.

Reply:
Now that you mentioned it WinUI default is dark mode. And it makes more sense to treat no value as zero. Will update this.